api/router/volume.go

```diff
@@ -37,21 +41,90 @@ func (v *VolumeRouter) Routes() []Route {
 }
 
 func (v *VolumeRouter) VolumeCreateHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	projectID := middleware.GetProjectIDFromContext(r.Context())
 
+	pcr := &types.VolumeCreateRequest{}
+	if err := render.Bind(r, pcr); err != nil {
+		render.Render(w, r, types.ErrHTTPBadRequest(err, "Failed to parse request"))
+		return
+	}
+
+	vol, err := v.uwService.Create(r.Context(), projectID, pcr.Name, pcr.Size)
+	if err != nil {
+		err = fmt.Errorf("failed to create volume, %w", err)
+		render.Render(w, r.WithContext(ctx), types.ErrHTTPError(err, "Failed to create volume"))
+	}
+
+	render.JSON(w, r, vol)
 }
 
 func (v *VolumeRouter) VolumeDeleteHandler(w http.ResponseWriter, r *http.Request) {
+	projectID := middleware.GetProjectIDFromContext(r.Context())
+
+	pcr := &types.VolumeDeleteRequest{}
+	if err := render.Bind(r, pcr); err != nil {
+		render.Render(w, r, types.ErrHTTPBadRequest(err, "Failed to parse request"))
+		return
+	}
+
+	err := v.uwService.Delete(r.Context(), projectID, pcr.IDOrName)
+	if err != nil {
+		err = fmt.Errorf("failed to delete volume, %w", err)
+		render.Render(w, r, types.ErrHTTPError(err, "Failed to delete volume"))
+		return
+	}
 
+	render.Status(r, 200)
 }
 
 func (v *VolumeRouter) VolumeGetHandler(w http.ResponseWriter, r *http.Request) {
+	projectID := middleware.GetProjectIDFromContext(r.Context())
+
+	pcr := &types.VolumeDeleteRequest{}
```

Comment:
Wrong type binding here. Should be `types.VolumeGetRequest`. This types actually doesn't exist so we don't need any binding at all.